ext/mysqli/mysqli_nonapi.c

```diff
@@ -1154,22 +1168,21 @@ PHP_FUNCTION(mysqli_begin_transaction)
 	zend_long		flags = TRANS_START_NO_OPT;
 	char *		name = NULL;
 	size_t			name_len = -1;
-	zend_bool	err = FALSE;
 
 	if (zend_parse_method_parameters(ZEND_NUM_ARGS(), getThis(), "O|ls!", &mysql_link, mysqli_link_class_entry, &flags, &name, &name_len) == FAILURE) {
 		RETURN_THROWS();
 	}
 	MYSQLI_FETCH_RESOURCE_CONN(mysql, mysql_link, MYSQLI_STATUS_VALID);
 	if (flags < 0) {
-		php_error_docref(NULL, E_WARNING, "Invalid value for parameter flags (" ZEND_LONG_FMT ")", flags);
-		err = TRUE;
+		zend_argument_value_error(ERROR_ARG_POS(2), "must be MYSQLI_TRANS_START_WITH_CONSISTENT_SNAPSHOT"
```

Comment:
maybe `one of` could be added for consistency with other messages?

Reply:
Good catch, fixed it :)